npe2/manifest/package_metadata.py

```diff
@@ -205,6 +205,9 @@ def from_dist_metadata(cls, meta: "email.message.Message") -> "PackageMetadata":
                 d[key] = value
         return cls.parse_obj(d)
 
+    def __hash__(self) -> int:
+        return id(self)
```

Comment:
This is simple and almost always gets the job done but it lacks the property that if `a==b` then `hash(a)==hash(b)`. When things go wrong, it's usually hard to diagnose so I feel like the use of `id()` should be explicitly done by the caller.

Not sure how feasible it is here, but I've used tuples to combine hashable values. Something along the lines of: `hash(tuple(thing.as_ordered_dict().items()))`